basyx.components/basyx.components.docker/basyx.components.AASServer/src/main/java/org/eclipse/basyx/components/aas/mongodb/MongoDBSubmodelAPI.java

```diff
@@ -557,19 +555,26 @@ public void addSubmodelElement(String idShortPath, ISubmodelElement elem) {
 
 	@SuppressWarnings("unchecked")
 	@Override
-	public Object getSubmodelElementFile(String idShortPath) {
+	public java.io.File getSubmodelElementFile(String idShortPath) {
 		try {
 			Map<String, Object> submodelElement = (Map<String, Object>) getSubmodelElement(idShortPath);
 			File fileSubmodelElement = File.createAsFacade(submodelElement);
 			GridFSBucket bucket = getGridFSBucket();
-			String fileName = fileSubmodelElement.getIdShort() + "." + fileSubmodelElement.getMimeType();
+			String fileName = constructFileName(fileSubmodelElement);
 			java.io.File file = new java.io.File(fileName);
 			FileOutputStream fileOutputStream;
 			fileOutputStream = new FileOutputStream(file);
 			bucket.downloadToStream(fileName, fileOutputStream);
 			return file;
 		} catch (FileNotFoundException e) {
-			throw new ProviderException("The File Submodel Element does not contain a File");
+			throw new ResourceNotFoundException("The File Submodel Element does not contain a File");
 		}
 	}
+
+	private String constructFileName(File file) {
+		String fileName;
+		Submodel sm = (Submodel) getSubmodel();
+		fileName = sm.getIdentification().getId() + "-" + file.getIdShort() + "." + file.getMimeType();
```

Comment:
This should be the complete idShort path, not only the idShort of the File SubmodelElement